Assets/HoloToolkit/Input/Scripts/Utilities/Interactions/TwoHandRotateLogic.cs

```diff
@@ -0,0 +1,142 @@
+            }
+            set
+            {
+                currentRotationConstraint = value;
+            }
+        }
+
+        private Vector3 ProjectHandlebarGivenConstraint(AxisConstraint constraint, Vector3 handlebarRotation, Transform manipulationRoot)
+        {
+            Vector3 result = handlebarRotation;
+            switch (constraint)
+            {
+                case AxisConstraint.XAxisOnly:
+                    result.x = 0;
+                    break;
+                case AxisConstraint.YAxisOnly:
+                    result.y = 0;
+                    break;
+                case AxisConstraint.ZAxisOnly:
+                    result.z = 0;
+                    break;
+            }
+            return CameraCache.Main.transform.TransformDirection(result);
+        }
+
+        private Vector3 GetHandlebarDirection(Dictionary<uint, Vector3> handsPressedMap, Transform manipulationRoot)
+        {
+            Assert.IsTrue(handsPressedMap.Count > 1);
+            var handsEnumerator = handsPressedMap.Values.GetEnumerator();
+            handsEnumerator.MoveNext();
+            var hand1 = handsEnumerator.Current;
+            handsEnumerator.MoveNext();
+            var hand2 = handsEnumerator.Current;
+
+            // We project the handlebar direction into camera space because otherwise when we move our body the handlebard will move even 
+            // though, relative to our heads, the handlebar is not moving.
+            hand1 = CameraCache.Main.transform.InverseTransformPoint(hand1);
+            hand2 = CameraCache.Main.transform.InverseTransformPoint(hand2);
+
+            return hand2 - hand1;
+        }
+
+        /// <summary>
+        /// Sets Axis Constraint for Rotation operation.
+        /// </summary>
+        /// <param name="rotationConstraint"></param>
+        public TwoHandRotateLogic(AxisConstraint rotationConstraint)
+        {
+            m_rotationConstraint = rotationConstraint;
+        }
+
+        /// <summary>
+        /// Initializes twohand system with controller/hand source info: 
+        /// the Dictionary collection already filled with controller/hand info,
+        /// and the Transform of the GameObject to be manipulated.
+        /// </summary>
+        /// <param name="handsPressedMap"></param>
+        /// <param name="manipulationRoot"></param>
+        public void Setup(Dictionary<uint, Vector3> handsPressedMap, Transform manipulationRoot)
+        {
+            currentRotationConstraint = m_rotationConstraint;
+            m_previousHandlebarRotation = GetHandlebarDirection(handsPressedMap, manipulationRoot);
+        }
+
+      
+        /// <summary>
+        /// Updates internal states based on current Controller/hand states.
+        /// </summary>
+        /// <param name="handsPressedMap"></param>
+        /// <param name="manipulationRoot"></param>
+        /// <param name="currentRotation"></param>
+        /// <returns></returns>
```

Comment:
Please be sure to document what is returned, semantically. Ex: "Quaternion representing ..."